src/app/shared/document/document-detail/document-detail.component.ts

```diff
@@ -0,0 +1,396 @@
+  }
+
+  private patchDocument() {
+    const updatedDocument = this.prepareUpdatedDocument();
+
+    if (updatedDocument) {
+      const documentUpdateAction = new DocumentsAction.UpdateData({ document: updatedDocument });
+
+      if (updatedDocument.newData && Object.getOwnPropertyNames(updatedDocument.newData).length > 0) {
+        const newAttributes = Object.keys(updatedDocument.newData).map(name => ({name, constraints: [], correlationId: updatedDocument.newData[name].correlationId}));
+
+        this.store.dispatch(new CollectionsAction.CreateAttributes(
+          {collectionId: this.documentModel.collectionId, attributes: newAttributes, nextAction: documentUpdateAction})
+        );
+      } else {
+        this.store.dispatch(documentUpdateAction);
+      }
+    }
+  }
+
+  public ngOnDestroy() {
+    this.subscriptions.unsubscribe();
+    this.patchDocument();
+  }
+
+  public addAttrRow() {
+    this.rows.push({ name: "", value: "", correlationId: CorrelationIdGenerator.generate(), warning: this.getEmptyWarning() });
+  }
+
+  public onRemoveRow(idx) {
+    if (this.rows[idx].name) {
+      const message = this.i18n(
+        {
+          id: 'document.detail.attribute.remove.confirm',
+          value: 'Are you sure you want to delete this row?'
+        });
+      const title = this.i18n({id: 'resource.delete.dialog.title', value: 'Delete?'});
+      const yesButtonText = this.i18n({id: 'button.yes', value: 'Yes'});
+      const noButtonText = this.i18n({id: 'button.no', value: 'No'});
+
+      this.notificationService.confirm(message, title, [
+        {text: yesButtonText, action: () => this.removeRow(idx), bold: false},
+        {text: noButtonText}
+      ]);
+    } else {
+      this.removeRow(idx);
+    }
+  }
+
+  private removeRow(idx) {
+    if (!this.rows[idx].name) {
+      this.rows.splice(idx, 1);
+    } else {
+      this.rows[idx].value = undefined;
+      this.rows[idx].remove = true;
+    }
+  }
+
+  public submitRowChange(idx, $event: string[]) {
+    const collision = this.getRowByName($event[0]);
+
+    if (collision && collision !== this.rows[idx]) {
+      this.rows[idx].warning = this.getCollisionWarning();
+    } else {
+      this.rows[idx].name = $event[0];
+      this.rows[idx].value = $event[1];
+
+      if (!$event[0]) {
+        this.rows[idx].warning = this.getEmptyWarning();
+      } else {
+        delete this.rows[idx].warning;
+      }
+    }
+  }
+
+  private getEmptyWarning() {
+    return this.i18n({ id: "shared.document.detail.attribute.empty", value: "The attribute name cannot be empty." })
+  }
+
+  private getCollisionWarning() {
+    return this.i18n({ id: "shared.document.detail.attribute.collision", value: "The attribute name is already used in this document." })
+  }
+
+  private getDocumentSummary() {
+    if (this.collection.defaultAttributeId) {
+      return this.documentModel.data[this.collection.defaultAttributeId];
+    }
+
+    if (this.collection.attributes.length > 0) {
```

Comment:
Check global method `getDefaultAttributeId` and use it here.

Reply:
I mean this method: https://github.com/Lumeer/web-ui/blob/devel/src/app/core/store/collections/collection.util.ts#L22.

And usage is then:
```
const attributeId = getDefaultAttibuteId(this.collection);
return this.documentModel.data[attributeId];
```

Reply:
This is a different functionality. It can happen that the first attribute of the collection is unused/unset in the document. And I am in search of a text that describes the displayed document. Maybe we should later add this function to a shared library.